File: engine/model/src/main/scala/tools/packet/foreigneffect/ShowRecovery.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowRecovery(private var _characterId: Int, private var _amount: Byte) extends PacketInput {
  def characterId: Int = _characterId

  def amount: Byte = _amount

  override def opcode(): SendOpcode = SendOpcode.SHOW_FOREIGN_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowRecovery.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowRecovery(int characterId, byte amount) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_FOREIGN_EFFECT;
   }
}